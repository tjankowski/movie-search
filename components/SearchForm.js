import React from "react";
import styles from "styles/SearchForm.module.scss";

export default function SearchForm({ query, onChange }) {
  return (
    <form className={styles.searchForm}>
      <input
        type="text"
        value={query}
        placeholder="Type movie title here"
        onChange={(e) => onChange(e.target.value)}
        className={styles.searchForm__input}
      />
      <button
        type="button"
        className={styles.searchForm__button}
        onClick={() => onChange(query)}
      >
        Search
      </button>
    </form>
  );
}
