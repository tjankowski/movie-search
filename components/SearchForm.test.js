import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";
import SearchForm from "./SearchForm";

describe("SearchForm", () => {
  it("should render", () => {
    const component = renderer.create(<SearchForm />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with query", () => {
    const component = renderer.create(<SearchForm query={"test"} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with query and onChange", () => {
    const onChange = jest.fn();
    const component = renderer.create(
      <SearchForm query={"test"} onChange={onChange} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledTimes(0);
  });

  it("should call onChange on value change", () => {
    const onChange = jest.fn();
    const { getByPlaceholderText } = render(
      <SearchForm query={"test"} onChange={onChange} />
    );
    expect(onChange).toHaveBeenCalledTimes(0);
    const input = getByPlaceholderText(/Type movie title here/i);
    fireEvent.change(input, { target: { value: "query" } });
    expect(onChange).toHaveBeenCalledWith("query");
  });

  it("should call onChange on button click", () => {
    const onChange = jest.fn();
    const { getByRole } = render(
      <SearchForm query={"test"} onChange={onChange} />
    );
    expect(onChange).toHaveBeenCalledTimes(0);
    const button = getByRole("button");
    fireEvent.click(button);
    expect(onChange).toHaveBeenCalledWith("test");
  });
});
