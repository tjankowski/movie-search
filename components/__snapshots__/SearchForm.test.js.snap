// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render 1`] = `
<form
  className="searchForm"
>
  <input
    className="searchForm__input"
    onChange={[Function]}
    placeholder="Type movie title here"
    type="text"
  />
  <button
    className="searchForm__button"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</form>
`;

exports[`SearchForm should render with query 1`] = `
<form
  className="searchForm"
>
  <input
    className="searchForm__input"
    onChange={[Function]}
    placeholder="Type movie title here"
    type="text"
    value="test"
  />
  <button
    className="searchForm__button"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</form>
`;

exports[`SearchForm should render with query and onChange 1`] = `
<form
  className="searchForm"
>
  <input
    className="searchForm__input"
    onChange={[Function]}
    placeholder="Type movie title here"
    type="text"
    value="test"
  />
  <button
    className="searchForm__button"
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</form>
`;
